src/client/remote_nodestream.ml

```diff
@@ -83,12 +83,19 @@ class remote_nodestream ((ic,oc) as conn) =
             end
         end
       in
+      let _should_reinit = ref false in
       let rec loop_parts () =
         Llio.input_int ic >>= function
         | (-2) -> Logger.info_f_ "loop_parts done"
         | 1 ->
           begin
             Logger.debug_f_ "loop_entries" >>= fun () ->
+            begin
+              if !_should_reinit
+              then tlog_coll # reinit()
```

Comment:
Why is this needed? Looks like it should be a concern of the tlogcollection itself?